pandas/_libs/groupby.pyx

```diff
@@ -433,5 +436,212 @@ def _group_add(floating[:, :] out,
 group_add_float32 = _group_add['float']
 group_add_float64 = _group_add['double']
 
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _group_prod(floating[:, :] out,
+                int64_t[:] counts,
+                floating[:, :] values,
+                const int64_t[:] labels,
+                Py_ssize_t min_count=0):
+    """
+    Only aggregates on axis=0
+    """
+    cdef:
+        Py_ssize_t i, j, N, K, lab, ncounts = len(counts)
+        floating val, count
+        ndarray[floating, ndim=2] prodx, nobs
```

Comment:
Can these be memoryviews?

Reply:
Probably but I wonder if we want them to be because they are assigned ndarrays a few lines below:

```
nobs = np.zeros_like(out)
prodx = np.ones_like(out)
```

Would you like me to declare them as memoryviews?

Reply:
Supposedly lookups, assignment, and slicing are slightly faster with memoryviews.  Since I think thats all we do with these objects and we are doing it in a loop, using memoryviews should be preferred.

(cython takes care of converting the ndarray to memoryview, so the `nobs = np.zeros_like(...)` lines don't need to be changed.